book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,195 @@
+
+```
+/// The funding stream receiver categories
+pub enum FundingStreamsReceivers {
+    /// Electric Coin Company
+    ECC,
+    /// ZCash Foundation
+    ZF,
+    /// Major Grants
+    MG,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_RECEIVERS_NUMERATORS: &[(i32, FundingStreamsReceivers)] = &[
+    (7, FundingStreamsReceivers::ECC),
+    (5, FundingStreamsReceivers::ZF),
+    (8, FundingStreamsReceivers::MG),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDINGSTREAM_DENOMINATOR: i32 = 100;
+
+/// Height intervals for funding streams on each network.
+pub enum FundingStreamsHeightIntervals {
+    /// For the Mainnet
+    Mainnet,
+    /// For the Testnet
+    Testnet,
+}
+
+/// Start and end Heights for funding streams 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_HEIGHT_INTERVALS: &[(Height, Height, FundingStreamsHeightIntervals)] = &[
+    (Height(1_046_400), Height(2_726_400), FundingStreamsHeightIntervals::Mainnet),
+    (Height(1_028_500), Height(2_796_000), FundingStreamsHeightIntervals::Testnet),
+];
+```
+
+## General subsidy
+
+The block subsidy and miner reward among other utility functions are inside the general subsidy category.
+
+- `block_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Total block subsidy.
+- `miner_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Miner portion.
+- `transaction_fees(&Block) -> Result<Amount<NonNegative>, Error>` - Sum of all the transaction fees.
+- `coinbase_sum_outputs(&Transaction) -> Result<Amount<NonNegative>, Error>` - Sum of all output values in the coinbase transaction.
+- `find_output_with_amount(&Transaction, Amount<NonNegative>) -> Vec<transparent::Output>` - Outputs where value equal to Amount.
+
+## Founders reward
+
+Only functions specific to calculation of founders reward.
+
+- `founders_reward(Height, Network) -> Result<Amount<NonNegative>, Error>` - Founders reward portion for this block or 0.
+- `founders_reward_address(Height, Network) -> Result<String, Error>` - Address of the receiver founder at this block, address is a `String` because it can be a transparent or sapling address(After Heartwood/ZIP-213). 
+
+## Funding streams
+
+Only functions specific to the calculation of funding streams.
+
+- `funding_stream(height, newtork) -> Result<Amount<NonNegative>, Error>` - Funding stream portion for this block or 0.
+- `funding_stream_address(height, network) -> Result<String, Error>` - Address of the funding stream receiver at this block. The same as founders reward the returned address is a `String`.
```

Comment:
Let's make a new enum for the return value, which contains either:
* `zebra_chain::transparent:Address::PayToScriptHash`
* `zebra_chain::sapling:Address`

Typed return values help avoid bugs - our goal is to make invalid states un-representable at compile time.

Reply:
It is not totally clear to me what we should return here. If `PayToScriptHash` is the only option for `founders_reward_addresses` as suggested at https://github.com/ZcashFoundation/zebra/pull/1129#discussion_r501453848 then here it will be the same. If we need to introduce a type then i think it haves to be the return of the 2 address functions.

Reply:
Why do you want to return the same type from both address functions?

We might need different return types for different functions:
* `funding_stream_address` are all `PayToScriptHash`, and they won't change before Canopy
* `funding_stream_address` contain transparent addresses, and might contain shielded addresses in future
  * the testnet addresses are all transparent `PayToScriptHash`
  * a future network upgrade could add shielded addresses (particularly for direct major grants payments)
  * see https://zips.z.cash/zip-0214#mainnet-recipient-addresses

Reply:
Ok, very clear now, thanks.

I modified the return type here https://github.com/ZcashFoundation/zebra/pull/1129/commits/ad80a6711837e1be222d92696c449a599c36fe03###